logdna/logdna.py

```diff
@@ -39,88 +43,93 @@ def __init__(self, key, options={}):
+            self.flusher = threading.Timer(interval, self.flush)
             self.flusher.start()
 
+    def cleaning_after_success(self):
+        del self.buf[:]
+        self.exception_flag = False
+        if self.flusher:
+            self.flusher.cancel()
+            self.flusher = None
+
+    def excpetion_actions(self, excpetion):
+       if self.flusher:
+           self.flusher.cancel()
+           self.flusher = None
+       self.exception_flag = True
+       if self.verbose in ['true', 'error', 'err', 'e']:
+           self.internalLogger.debug('Error sending logs %s', excpetion)
+
+    # do not call without aquiring the lock
+    def send_request(self):
+       self.buf = self.buf + self.secondary
+       self.secondary = []
+       data = {'e': 'ls', 'ls': self.buf}
+       try:
+           res = requests.post(
+               url=self.url,
+               json=data,
+               auth=('user', self.key),
+               params={
+                   'hostname': self.hostname,
+                   'ip': self.ip,
+                   'mac': self.mac if self.mac else None,
+                   'tags': self.tags if self.tags else None},
+               stream=True,
+               timeout=self.request_timeout)
+           res.raise_for_status()
+           # when no RequestException happened
+           self.cleaning_after_success()
+       except requests.exceptions.RequestException as e:
+           self.excpetion_actions(e)
+
     def flush(self):
-        if not self.buf or len(self.buf) < 0:
-            return
-        data = {'e': 'ls', 'ls': self.buf}
-        try:
-            # Ensure we have the lock when flushing
-            if not self.lock.acquire(blocking=False):
-                if not self.flusher:
-                    self.flusher = threading.Timer(1, self.flush)
-                    self.flusher.start()
-            else:
-                self.session.post(
-                    url=self.url,
-                    json=data,
-                    auth=('user', self.key),
-                    params={
-                        'hostname': self.hostname,
-                        'ip': self.ip,
-                        'mac': self.mac if self.mac else None,
-                        'tags': self.tags if self.tags else None},
-                    stream=True,
-                    timeout=self.request_timeout)
-                self.buf = []
-                self.buf_byte_length = 0
-                if self.flusher:
-                    self.flusher.cancel()
-                    self.flusher = None
-                self.lock.release()
-                # Ensure messages that could've dropped are appended back onto buf
-                self.buf = self.buf + self.secondary
-                self.secondary = []
-        except requests.exceptions.RequestException as e:
-            if self.flusher:
-                self.flusher.cancel()
-                self.flusher = None
+        if len(self.buf) == 0: return
```

Comment:
Inline `if`s are unusual in Python, please move to next line. I also think `if not self.buf` is better but I don't feel strongly about that.